### PR TITLE
Standardize env loading for enclaves

### PR DESCRIPTION
**Pull request by lfarrel6**
# Why

There was significant drift in how the env vars were loaded for Enclaves based on whether they had TLS enabled or not. With TLS enabled, env var loading was handled when fetching the TLS certificate. For TLS Disabled, env vars were loaded in isolation. The drift between both flows has made it awkward to make the init flow more resilient. This change aims to centralise all env var loading into a shared client.

# How

- Replace old Environment module with a new EnvironmentLoader which uses distinct states to expose the appropriate requests to the data plane
- Update the data plane to load the environment variables immediately so the flow is consistent across both TLS modes
- Pass the environment loader to the server builder to pull in the certs
- Move notify shutdown trait into the shared lib so it can be applied to the control plane tasks.

### CODE REVIEW

File: data-plane/src/server/tls/tls_server.rs [hunk 1]
```diff
@@ -5,29 +5,24 @@ use once_cell::sync::OnceCell;
...
-use std::thread;
-use std::time::Duration;
 use tokio_rustls::rustls::server::WantsServerCert;
 use tokio_rustls::rustls::ConfigBuilder;
 use tokio_rustls::rustls::ServerConfig;
 use tokio_rustls::server::TlsStream;
 use tokio_rustls::TlsAcceptor;
 
-use super::inter_ca_retreiver;
+// use super::inter_ca_retreiver;
```
Comment (hanneary) [inline data-plane/src/server/tls/tls_server.rs:17]: can be removed?

File: data-plane/src/server/tls/tls_server.rs [hunk 2]
```diff
@@ -80,50 +75,27 @@ impl<S: Listener + Send + Sync> WantsCert<S> {
...
-    pub async fn with_attestable_cert(self) -> ServerResult<TlsServer<S>> {
+    pub async fn with_attestable_cert(
+        self,
+        env_loader: EnvironmentLoader<NeedCert>,
+    ) -> ServerResult<TlsServer<S>> {
         log::info!("Creating TLSServer with attestable cert");
-        let (ca_cert, ca_private_key) = Self::get_ca_with_retry().await;
-        log::debug!("Received intermediate CA from cert provisioner. Using it with TLS Server.");
+
+        let (env_loader, inter_ca_cert, inter_ca_key_pair) = env_loader.load_cert().await.unwrap();
```
Comment (hanneary) [inline data-plane/src/server/tls/tls_server.rs:84]: do we wanna unwrap here?

File: data-plane/src/server/server.rs [hunk 3]
```diff
@@ -29,14 +30,18 @@ use super::layers::{
...
+) where
     TlsError: From<<L as Listener>::Error>,
     <L as Listener>::Connection: ProxiedConnection + 'static,
 {
     let mut server = TlsServerBuilder::new()
         .with_server(tcp_server)
-        .with_attestable_cert()
+        .with_attestable_cert(env_loader)
         .await
         .expect("Failed to create tls server");
```
Comment (hanneary) [inline data-plane/src/server/server.rs:46]: we should prob give error context here

**Review by hanneary — COMMENTED** [review]
lgtm just a few qs

File: data-plane/src/server/tls/tls_server.rs (hunk 1 above)
Comment (lfarrel6) [reply data-plane/src/server/tls/tls_server.rs:17]: yep, missed it

**Review by hanneary — APPROVED** [review]

**Pull request merged**